plugins/repoquery.py

```diff
@@ -348,19 +378,46 @@ def run(self, args):
             q = self.base.sack.query().filter(pkg=pkg_list)
         fmt_fn = build_format_fn(self.opts)
         if self.opts.tree:
-            if not self.opts.whatrequires and \
-                "%{requires}" not in self.opts.queryformat and \
-                "%{conflicts}" not in self.opts.queryformat and \
-                "%{obsoletes}" not in self.opts.queryformat:
+            if not self.opts.whatrequires and not self.opts.packageatr:
                 raise dnf.exceptions.Error(
-                    _("--tree requires either of these options: --whatrequires --requires --conflicts --obsoletes\n"
-                      "Example: dnf repoquery --whatrequires audiofile --tree"))
+                    _("No switch specified\nusage: dnf repoquery [--whatrequires|"
+                        "--requires|--conflicts|--obsoletes|--enhances|--suggest|"
+                        "--provides|--suplements|--recommends] [key] [--tree]\n\n"
+                        "description:\n  For the given packages print a tree of the packages."))
             self.tree_seed(q, orquery, self.opts)
             return
-        if self.opts.resolve:
-            self.show_resolved_packages(q, fmt_fn, self.opts)
+
+        pkgs = set()
+        if self.opts.packageatr:
+            for pkg in q.run():
+                rels = getattr(pkg, self.opts.packageatr)
+                for rel in rels:
+                    pkgs.add(str(rel))
         else:
-            self.show_packages(q, fmt_fn)
+            for pkg in q.run():
+                po = PackageWrapper(pkg)
+                try:
+                    pkgs.add(fmt_fn(po))
+                except AttributeError as e:
+                    # catch that the user has specified attributes
+                    # there don't exist on the dnf Package object.
+                    raise dnf.exceptions.Error(str(e))
+        if self.opts.resolve:
+            # find the providing packages and show them
+            query = self.filter_repo_arch(self.opts, self.base.sack.query().available())
+            providers = self.by_provides(self.base.sack, list(pkgs),
+                                         query)
+            pkgs = set()
+            for pkg in providers.latest().run():
+                po = PackageWrapper(pkg)
+                try:
+                    pkgs.add(fmt_fn(po))
+                except AttributeError as e:
+                    # catch that the user has specified attributes
+                    # there don't exist on the dnf Package object.
+                    raise dnf.exceptions.Error(str(e))
+        for pkg in sorted(pkgs, key=lambda s: str(s)):
```

Comment:
@max9631 , change it to `for pkg in sorted(pkgs, key=i18n.ucd)` and see if it works.